### ISSUE TITLE
Unable to Install Open3D v18.0, Maximum Supported Version is 13.0

### ISSUE THREAD

**Issue by zygiauzas** [coleygroup/shepherd#2]
I followed the installation instructions, but I am encountering an issue with the Open3D library when trying to install version 18.0. My system setup allows me to install up to version 13.0, but I cannot reach the required v18.0. With v13, certain shepherd functions are not being imported. This seems to be a limitation in the installation process, possibly related to my environment or dependencies. Could you provide guidance on how to resolve this or any known issues preventing the installation of version 18.0?

**Comment by kentoabeywardane** [comment]
Hi, thanks for reaching out. We have not run into this issue ourselves. Would you be able to provide more details about your python and numpy versions? Since numpy is being upgraded after the installation of open3d, it is possible that there could be a mismatch in required version numbers.

These are the required dependencies of Open3D 0.18.0 (`pip show open3d`):
addict, configargparse, dash, ipywidgets, matplotlib, nbformat, numpy, numpy, pandas, pillow, pyquaternion, pyyaml, scikit-learn, tqdm, werkzeug

It may be helpful to look through to see if there are any mismatched versions from your environment to ours for these dependencies.

**Comment by zygiauzas** [comment]
Hi,

Thank you for your prompt response and the details regarding Open3D's dependencies. Here's the information you requested along with some additional context:

My Setup:
Python Version: 3.8
NumPy Version: 1.24.4
Open3D Version: 0.13.0

Thank you for helping out. Below is the list of dependencies, that I installed based on the provided commands.

Dependencies as Reported by pip :
name: shepherd
channels:
  - pyg
  - pytorch
  - merv
  - defaults
  - https://repo.anaconda.com/pkgs/main
  - https://repo.anaconda.com/pkgs/r
  - conda-forge
dependencies:
  - _libgcc_mutex=0.1
  - _openmp_mutex=4.5
  - blas=1.0
  - brotli-python=1.0.9
  - bzip2=1.0.8
  - c-ares=1.19.1
  - ca-certificates=2024.11.26
  - certifi=2024.8.30
  - charset-normalizer=3.3.2
  - cudatoolkit=11.3.1
  - dftd4=3.7.0
  - envvar-pythonnousersite-true=0.1.0
  - ffmpeg=4.3
  - freetype=2.12.1
  - gmp=6.2.1
  - gnutls=3.6.15
  - h5py=3.11.0
  - hdf5=1.12.1
  - idna=3.7
  - intel-openmp=2023.1.0
  - jinja2=3.1.4
  - joblib=1.4.2
  - jpeg=9e
  - krb5=1.20.1
  - lame=3.100
  - lcms2=2.12
  - ld_impl_linux-64=2.40
  - lerc=3.0
  - libblas=3.9.0
  - libcurl=8.9.1
  - libdeflate=1.17
  - libedit=3.1.20230828
  - libev=4.33
  - libffi=3.4.4
  - libgcc=14.2.0
  - libgcc-ng=14.2.0
  - libgfortran=14.2.0
  - libgfortran-ng=14.2.0
  - libgfortran5=14.2.0
  - libgomp=14.2.0
  - libiconv=1.16
  - libidn2=2.3.4
  - liblapack=3.9.0
  - libnghttp2=1.57.0
  - libpng=1.6.39
  - libssh2=1.11.1
  - libstdcxx-ng=11.2.0
  - libtasn1=4.19.0
  - libtiff=4.5.1
  - libunistring=0.9.10
  - libwebp-base=1.3.2
  - lz4-c=1.9.4
  - markupsafe=2.1.3
  - mctc-lib=0.3.2
  - mkl=2023.1.0
  - mkl-service=2.4.0
  - mkl_fft=1.3.8
  - mkl_random=1.2.4
  - multicharge=0.3.0
  - ncurses=6.4
  - nettle=3.7.3
  - openh264=2.1.1
  - openjpeg=2.5.2
  - openssl=3.0.15
  - packaging=24.1
  - pillow=10.4.0
  - platformdirs=3.10.0
  - pooch=1.7.0
  - psutil=5.9.0
  - pyg=2.2.0
  - pyparsing=3.1.2
  - pysocks=1.7.1
  - python=3.8.20
  - pytorch=1.12.1
  - pytorch-cluster=1.6.0
  - pytorch-mutex=1.0
  - pytorch-scatter=2.1.0
  - pytorch-sparse=0.6.16
  - readline=8.2
  - requests=2.32.3
  - scikit-learn=1.3.0
  - scipy=1.10.1
  - simple-dftd3=1.2.1
  - sqlite=3.45.3
  - tbb=2021.8.0
  - tblite=0.4.0
  - threadpoolctl=3.5.0
  - tk=8.6.14
  - toml-f=0.4.2
  - torchaudio=0.12.1
  - torchvision=0.13.1
  - tqdm=4.66.5
  - typing_extensions=4.11.0
  - urllib3=2.2.3
  - wheel=0.44.0
  - xtb=6.7.1
  - xz=5.4.6
  - zlib=1.2.13
  - zstd=1.5.6
  - pip:
      - absl-py==2.1.0
      - addict==2.4.0
      - aiofiles==22.1.0
      - aiohappyeyeballs==2.4.4
      - aiohttp==3.10.11
      - aiosignal==1.3.1
      - aiosqlite==0.20.0
      - anyio==4.5.2
      - argon2-cffi==23.1.0
      - argon2-cffi-bindings==21.2.0
      - arrow==1.3.0
      - asttokens==3.0.0
      - async-timeout==5.0.1
      - attrs==24.2.0
      - babel==2.16.0
      - backcall==0.2.0
      - beautifulsoup4==4.12.3  
      - bleach==6.1.0
      - cachetools==5.5.0
      - cffi==1.17.1
      - comm==0.2.2
      - contourpy==1.1.1
      - cycler==0.12.1
      - debugpy==1.8.9
      - decorator==5.1.1
      - defusedxml==0.7.1
      - deprecation==2.1.0
      - e3nn==0.5.4
      - entrypoints==0.4
      - exceptiongroup==1.2.2
      - executing==2.1.0
      - fastjsonschema==2.21.1
      - fonttools==4.55.1
      - fqdn==1.5.1
      - frozenlist==1.5.0
      - fsspec==2024.10.0
      - google-auth==2.36.0
      - google-auth-oauthlib==1.0.0
      - grpcio==1.68.1
      - importlib-metadata==8.5.0
      - importlib-resources==6.4.5
      - ipykernel==6.29.5
      - ipython==8.12.3
      - ipython-genutils==0.2.0
      - ipywidgets==8.1.5
      - isoduration==20.11.0
      - jedi==0.19.2
      - json5==0.10.0
      - jsonpointer==3.0.0
      - jsonschema==4.23.0
      - jsonschema-specifications==2023.12.1
      - jupyter-client==7.4.9
      - jupyter-core==5.7.2
      - jupyter-events==0.10.0
      - jupyter-packaging==0.12.3
      - jupyter-server==2.14.2
      - jupyter-server-fileid==0.9.3
      - jupyter-server-terminals==0.5.3
      - jupyter-server-ydoc==0.8.0
      - jupyter-ydoc==0.2.5
      - jupyterlab==3.6.8
      - jupyterlab-pygments==0.3.0
      - jupyterlab-server==2.27.3
      - jupyterlab-widgets==3.0.13
      - kiwisolver==1.4.7
      - lightning-utilities==0.11.9
      - markdown==3.7
      - matplotlib==3.7.5
      - matplotlib-inline==0.1.7
      - mistune==3.0.2
      - mpmath==1.3.0
      - multidict==6.1.0
      - nbclassic==1.1.0
      - nbclient==0.10.1
      - nbconvert==7.16.4
      - nbformat==5.10.4
      - nest-asyncio==1.6.0
      - notebook==6.5.7
      - notebook-shim==0.2.4
      - numpy==1.24.4
      - oauthlib==3.2.2
      - open3d==0.13.0
      - opt-einsum==3.4.0
      - opt-einsum-fx==0.1.4
      - overrides==7.7.0
      - pandas==2.0.3
      - pandocfilters==1.5.1
      - parso==0.8.4
      - pexpect==4.9.0
      - pickleshare==0.7.5
      - pip==24.0
      - pkgutil-resolve-name==1.3.10
      - prometheus-client==0.21.0
      - prompt-toolkit==3.0.48
      - propcache==0.2.0
      - protobuf==5.29.0
      - ptyprocess==0.7.0
      - pure-eval==0.2.3
      - pyasn1==0.6.1
      - pyasn1-modules==0.4.1
      - pycparser==2.22
      - pydeprecate==0.3.2
      - pygments==2.18.0
      - python-dateutil==2.9.0.post0
      - python-json-logger==2.0.7
      - pytorch-lightning==1.6.3
      - pytz==2024.2
      - pyyaml==6.0.2
      - pyzmq==26.2.0
      - rdkit==2024.3.2
      - referencing==0.35.1
      - requests-oauthlib==2.0.0
      - rfc3339-validator==0.1.4
      - rfc3986-validator==0.1.1
      - rpds-py==0.20.1
      - rsa==4.9
      - send2trash==1.8.3
      - setuptools==75.3.0
      - six==1.16.0
      - sniffio==1.3.1
      - soupsieve==2.6
      - stack-data==0.6.3
      - sympy==1.13.3
      - tensorboard==2.14.0
      - tensorboard-data-server==0.7.2
      - terminado==0.18.1
      - tinycss2==1.4.0
      - tomli==2.2.1
      - tomlkit==0.13.2
      - torchmetrics==1.5.2
      - tornado==6.4.2
      - traitlets==5.14.3
      - types-python-dateutil==2.9.0.20241003
      - tzdata==2024.2
      - uri-template==1.3.0
      - wcwidth==0.2.13
      - webcolors==24.8.0
      - webencodings==0.5.1
      - websocket-client==1.8.0
      - werkzeug==3.0.6
      - widgetsnbextension==4.0.13
      - y-py==0.6.2
      - yarl==1.15.2
      - ypy-websocket==0.8.4
      - zipp==3.20.2

**Comment by kentoabeywardane** [comment]
Hi, sorry for the delay on this.

I'm not entirely sure what could be wrong in the environment since there seems to be minimal differences between the versions you have and the one we use.

Could you expand on what shepherd functions fail due to Open3D being the wrong version?

If you havn't already, please try to upgrade the versions again, or retry the installation process.
